src/raftstore/store/peer.rs

```diff
@@ -1102,6 +1125,7 @@ impl Peer {
                 .advance_apply(apply_state.get_applied_index());
         }
         self.mut_store().apply_state = apply_state;
+        let progress_to_be_updated = self.mut_store().applied_index_term != applied_index_term;
```

Comment:
do you need to expire the remote lease here?

Reply:
can you tell me where to expire?

Reply:
It will first call `on_role_changed`, and if it becomes a follower, it expires its lease. Expiring lease also causes a remote lease expired.